lib/mpesa/resources/token.rb

```diff
@@ -25,7 +25,9 @@ def cache_token
     end
 
     def call
-      Object.new get_request(url: 'oauth/v1/generate?grant_type=client_credentials', basic_auth: true).body
+      res = get_request(url: 'oauth/v1/generate?grant_type=client_credentials', basic_auth: true)
+      raise res.reason_phrase if res.body.empty?
```

Comment:
```suggestion
      raise Error, res.reason_phrase if res.body.empty?
```
This will use the [custom error](https://github.com/gathuku/mpesa/blob/591f6290c2e64d242c00e6d649f403e1fc2ac282/lib/mpesa/error.rb#L4). The constant is [autoloaded](https://github.com/gathuku/mpesa/blob/591f6290c2e64d242c00e6d649f403e1fc2ac282/lib/mpesa.rb#L7) and will be available to code inside `Mpesa` module. This will raise `Mpesa::Error`